src/app/hat-claim/hat-claim/hat-claim.component.ts

```diff
@@ -46,7 +46,11 @@ export class HatClaimComponent implements OnInit {
   }
 
   nextStep(): void {
-    if (this.step === 3) {
+    if (this.step === 1) {
+      if (this.email) {
```

Comment:
should avoid nested if statements whenever possible. This could be replaced with `if (this.step === 1 && this.email)`